services/src/test/java/org/exoplatform/onlyoffice/OnlyofficeEditorServiceTest.java

```diff
@@ -110,16 +111,93 @@ protected NodeImpl createDocument(String title, String type, String data, Boolea
     return node;
   }
 
-  protected void startSessionAs(String user) throws Exception {
-    HashSet<MembershipEntry> memberships = new HashSet<MembershipEntry>();
-    memberships.add(new MembershipEntry("/platform/administrators"));
-    Identity identity = new Identity(user, memberships);
-    ConversationState state = new ConversationState(identity);
-    state.setAttribute(ConversationState.SUBJECT, identity.getSubject());
-    ConversationState.setCurrent(state);
-    SessionProvider provider = new SessionProvider(state);
-    sessionProviderService.setSessionProvider(null, provider);
-    session = provider.getSession(WORKSPACE_NAME, repositoryService.getCurrentRepository());
+  /**
+   * Test add listener
+   */
+  @Test
+  public void testCallListenerOnCreateWhenAddingListener() throws Exception {
+    // Given
+    startSessionAs("john");
+    Node node = createDocument("Test Document.docx", "nt:file", "testContent", true);
+    AtomicBoolean onCreateCalled = new AtomicBoolean(false);
+    OnlyofficeEditorListener listener = new OnlyofficeEditorListener() {
+      @Override
+      public void onCreate(DocumentStatus status) {
+        onCreateCalled.set(true);
+      }
+
+      @Override
+      public void onGet(DocumentStatus status) {
+
+      }
+
+      @Override
+      public void onJoined(DocumentStatus status) {
+
+      }
+
+      @Override
+      public void onLeaved(DocumentStatus status) {
+
+      }
+
+      @Override
+      public void onSaved(DocumentStatus status) {
+
+      }
+
+      @Override
+      public void onError(DocumentStatus status) {
+
+      }
+    };
+    editorService.addListener(listener);
+
+    // When
+    editorService.createEditor("http", "127.0.0.1", 8080, "john", null, node.getUUID());
+
+    // Then
+    assertTrue(onCreateCalled.get());
+    node.remove();
+  }
+
+  /**
+   * Test create editor
+   */
+  @Test
+  public void testCreateEditor() throws Exception {
```

Comment:
Which case of the createEditor does this test ? We should make the method name more precise.